tests/v2/actions/rbac/rbac.go

```diff
@@ -0,0 +1,68 @@
+package rbac
+
+import (
+	"strings"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/rbac"
+	"github.com/rancher/shepherd/extensions/users"
+)
+
+type Role string
+
+const (
+	RestrictedAdmin           Role = "restricted-admin"
+	StandardUser              Role = "user"
+	ClusterOwner              Role = "cluster-owner"
+	ClusterMember             Role = "cluster-member"
+	ProjectOwner              Role = "project-owner"
+	ProjectMember             Role = "project-member"
+	CreateNS                  Role = "create-ns"
+	ReadOnly                  Role = "read-only"
+	CustomManageProjectMember Role = "projectroletemplatebindings-manage"
+	ActiveStatus                   = "active"
+	ForbiddenError                 = "403 Forbidden"
+	DefaultNamespace               = "fleet-default"
+)
+
+func (r Role) String() string {
+	return string(r)
+}
+
+// CreateUserAndAddToClusterRole creates a user based on the global role and then adds the user to cluster with provided permissions.
+func CreateUserAndAddToClusterRole(adminClient *rancher.Client, globalRole, clusterRole string, cluster *management.Cluster, project *management.Project) (*management.User, *rancher.Client, error) {
```

Comment:
Since this also involves adding global roles, I've made it addClusterRole instead of just role which could mean both global and cluster related permissions.

Reply:
For project, it is a little misleading though

Reply:
Had an offline discussion and updated based on that.